src/js/actions/layers.js

```diff
@@ -1377,45 +1382,154 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard \
+     * 
+     * @private 
+     * @param {Immutable.List.<Layer>} artboards
+     * @return {Layer}
+     */
+    var _findRightMostArtboard = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+            if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                return currentLayer;
+            } else {
+                return selectedLayer;
+            }
+        }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     *
+     * @private
+     * @param {object} specs 
+     * @param {Immutable.List.<Layer>} artboards
+     * @param {Layer} layerRef
+     * @return {{finalBounds: Bounds, layerRef: Layer}}
+     */
+    var _getBoundsFromSpecs = function (specs, artboards, layerRef) {
+        var preset = specs.preset,
```

Comment:
I think it would make more sense to group the decls and initializations as follows:
```javascript
        var preset = specs.preset,
            index = _.findIndex(templates, function (obj) {
                return obj.preset === preset;
            });

        var height = templates[index].height;
            width = templates[index].width,
            finalBounds;
```